diff --git a/meson.build b/meson.build
index a9e2c1f..391c1d5 100644
--- a/meson.build
+++ b/meson.build
@@ -71,5 +71,7 @@ subdir('data')
 if not get_option('disable-help')
     subdir('help')
 endif
-subdir('tests')
-subdir('tests/validate-tests')
+if get_option('enable-tests')
+  subdir('tests')
+  subdir('tests/validate-tests')
+endif
diff --git a/meson_options.txt b/meson_options.txt
index 05283fa..fa85985 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -1 +1,2 @@
 option('disable-help', type : 'boolean', value : false)
+option('enable-tests', type : 'boolean', value : false)
