diff --git a/Telegram/SourceFiles/window/window_filters_menu.cpp b/Telegram/SourceFiles/window/window_filters_menu.cpp
index 1e71a84b..21efd394 100644
--- a/Telegram/SourceFiles/window/window_filters_menu.cpp
+++ b/Telegram/SourceFiles/window/window_filters_menu.cpp
@@ -285,8 +285,7 @@ base::unique_qptr<Ui::SideBarButton> FiltersMenu::prepareButton(
 			const auto chats = state.chats;
 			const auto chatsMuted = state.chatsMuted;
 			const auto muted = (chatsMuted + state.marksMuted);
-			const auto count = (chats + state.marks)
-				- (includeMuted ? 0 : muted);
+			const auto count = 0;
 			const auto string = !count
 				? QString()
 				: (count > 999)
