diff --git a/Telegram/SourceFiles/window/window_filters_menu.cpp b/Telegram/SourceFiles/window/window_filters_menu.cpp
index 88a940f..c44545b 100644
--- a/Telegram/SourceFiles/window/window_filters_menu.cpp
+++ b/Telegram/SourceFiles/window/window_filters_menu.cpp
@@ -201,7 +201,7 @@ void FiltersMenu::refresh() {
 		return;
 	}
 	const auto oldTop = _scroll.scrollTop();
-	const auto reorderAll = premium();
+	const auto reorderAll = true;
 	if (!_list) {
 		setupList();
 	}
@@ -546,7 +546,7 @@ void FiltersMenu::applyReorder(
 
 	const auto filters = &_session->session().data().chatsFilters();
 	const auto &list = filters->list();
-	if (!premium()) {
+	if (false) {
 		if (list[0].id() != FilterId()) {
 			filters->moveAllToFront();
 		}
