diff --git a/common/flatpak-context-private.h b/common/flatpak-context-private.h
index d9a7977..d83316d 100644
--- a/common/flatpak-context-private.h
+++ b/common/flatpak-context-private.h
@@ -57,6 +57,7 @@ typedef enum {
   FLATPAK_CONTEXT_DEVICE_ALL         = 1 << 1,
   FLATPAK_CONTEXT_DEVICE_KVM         = 1 << 2,
   FLATPAK_CONTEXT_DEVICE_SHM         = 1 << 3,
+  FLATPAK_CONTEXT_DEVICE_ALSA        = 1 << 4,
 } FlatpakContextDevices;
 
 typedef enum {
diff --git a/common/flatpak-context.c b/common/flatpak-context.c
index 3865380..8acb50e 100644
--- a/common/flatpak-context.c
+++ b/common/flatpak-context.c
@@ -70,6 +70,7 @@ const char *flatpak_context_devices[] = {
   "all",
   "kvm",
   "shm",
+  "alsa",
   NULL
 };
 
diff --git a/common/flatpak-run.c b/common/flatpak-run.c
index 8fa8c0e..d3c5943 100644
--- a/common/flatpak-run.c
+++ b/common/flatpak-run.c
@@ -1751,6 +1751,13 @@ flatpak_run_add_environment_args (FlatpakBwrap    *bwrap,
                                   "--bind", shared_dev_shm, "/dev/shm",
                                   NULL);
         }
+
+      if (context->devices & FLATPAK_CONTEXT_DEVICE_ALSA)
+        {
+          g_debug ("Allowing ALSA access");
+          if (g_file_test ("/dev/snd", G_FILE_TEST_EXISTS))
+            flatpak_bwrap_add_args (bwrap, "--dev-bind", "/dev/snd", "/dev/snd", NULL);
+        }
     }
 
   exports = flatpak_context_get_exports_full (context,
