diff -ur elinks-0.12-20090101/src/session/task.c /home/sbar/progs/elinks-0.12-20090101/src/session/task.c
--- elinks-0.12-20090101/src/session/task.c	2009-01-01 02:40:03.000000000 +0300
+++ /home/sbar/progs/elinks-0.12-20090101/src/session/task.c	2009-01-05 17:09:36.000000000 +0300
@@ -271,12 +271,15 @@
 		mem_free_if(uristring);
 	}
 
+	if (task_type != TASK_FORWARD)
 	msg_box(ses->tab->term, getml(task, (void *) NULL), MSGBOX_FREE_TEXT,
 		N_("Warning"), ALIGN_CENTER,
 		message,
 		task, 2,
 		MSG_BOX_BUTTON(N_("~Yes"), post_yes, B_ENTER),
 		MSG_BOX_BUTTON(N_("~No"), post_no, B_ESC));
+	else
+		post_yes(task);
 }
