diff --git a/src/monitor/glwidget.cpp b/src/monitor/glwidget.cpp
index 06ecf5a84..7d5e7d53c 100644
--- a/src/monitor/glwidget.cpp
+++ b/src/monitor/glwidget.cpp
@@ -704,10 +704,10 @@ void VideoWidget::slotZoom(bool zoomIn)
 {
     if (zoomIn) {
         if (m_zoom < 12.0f) {
-            setZoom(m_zoom * 1.2);
+            setZoom(m_zoom * 1.05);
         }
     } else if (m_zoom > 0.2f) {
-        setZoom(m_zoom / 1.2f);
+        setZoom(m_zoom / 1.05f);
     }
 }
 
