*** a/Telegram/SourceFiles/data/components/sponsored_messages.cpp	2024-11-05 11:45:43.000000000 +0300
--- b/Telegram/SourceFiles/data/components/sponsored_messages.cpp	2024-11-12 15:36:57.080621183 +0300
***************
*** 195,205 ****
  }
  
  bool SponsoredMessages::canHaveFor(not_null<History*> history) const {
- 	if (history->peer->isChannel()) {
- 		return true;
- 	} else if (const auto user = history->peer->asUser()) {
- 		return user->isBot();
- 	}
  	return false;
  }
  
--- 195,200 ----
