--- makeresolvedeb_1.6.4_multi.sh.orig	2023-09-30 08:42:23.794166730 +0100
+++ makeresolvedeb.sh	2023-09-30 12:38:17.423548867 +0100
@@ -158,8 +158,8 @@
     copy_object ${UNPACK_DIR}/libs/libBlackmagicRawAPI.so ${RESOLVE_BASE_DIR}/bin/
 
     # Install panel API library
-    create_directory ${DEB_DIR}/usr/lib/
-    extract_tgz ${UNPACK_DIR}/share/panels/dvpanel-framework-linux-x86_64.tgz ${DEB_DIR}/usr/lib/ libDaVinciPanelAPI.so
+    create_directory ${DEB_DIR}/usr/#LIBDIR#/
+    extract_tgz ${UNPACK_DIR}/share/panels/dvpanel-framework-linux-x86_64.tgz ${DEB_DIR}/usr/#LIBDIR#/ libDaVinciPanelAPI.so
 
     # Add postinst commands
     cat >> ${DEB_DIR}/DEBIAN/postinst <<EOF
@@ -215,8 +215,8 @@
     copy_object ${UNPACK_DIR}/share/default_cm_config.bin ${RESOLVE_BASE_DIR}/share/
 
     # Extract panel API library
-    create_directory ${DEB_DIR}/usr/lib
-    extract_tgz ${UNPACK_DIR}/share/panels/dvpanel-framework-linux-x86_64.tgz ${DEB_DIR}/usr/lib libDaVinciPanelAPI.so
+    create_directory ${DEB_DIR}/usr/#LIBDIR#/
+    extract_tgz ${UNPACK_DIR}/share/panels/dvpanel-framework-linux-x86_64.tgz ${DEB_DIR}/usr/#LIBDIR#/ libDaVinciPanelAPI.so
 
     # BlackmagicRawAPI fixes
     create_directory ${RESOLVE_BASE_DIR}/bin/BlackmagicRawAPI/
@@ -278,9 +278,9 @@
     copy_object ${UNPACK_DIR}/share/default_cm_config.bin ${RESOLVE_BASE_DIR}/share/
 
     # Extract panel API library
-    create_directory ${DEB_DIR}/usr/lib
-    extract_tgz ${UNPACK_DIR}/share/panels/dvpanel-framework-linux-x86_64.tgz ${DEB_DIR}/usr/lib libDaVinciPanelAPI.so
-    extract_tgz ${UNPACK_DIR}/share/panels/dvpanel-framework-linux-x86_64.tgz ${DEB_DIR}/usr/lib libFairlightPanelAPI.so
+    create_directory ${DEB_DIR}/usr/#LIBDIR#/
+    extract_tgz ${UNPACK_DIR}/share/panels/dvpanel-framework-linux-x86_64.tgz ${DEB_DIR}/usr/#LIBDIR#/ libDaVinciPanelAPI.so
+    extract_tgz ${UNPACK_DIR}/share/panels/dvpanel-framework-linux-x86_64.tgz ${DEB_DIR}/usr/#LIBDIR#/ libFairlightPanelAPI.so
 
     # BlackmagicRawAPI fixes for 17.0 & 17.1
     if [[ "$RESOLVE_VERSION" == 17.0* ]] || [[ "$RESOLVE_VERSION" == 17.1* ]];
@@ -633,8 +633,8 @@
 
 # Verify that we have the commands we need
 check_command tar
-check_command fakeroot
-check_command dpkg-deb
+#check_command fakeroot
+#check_command dpkg-deb
 echo
 
 # Create destination directories
